digitaltwins/device/src/digital_twin_client.ts

```diff
@@ -589,157 +586,31 @@ export class DigitalTwinClient {
-    //
-    // Set the timeout for timeoutForDisconnects seconds, which should be long enough to go through this entire sequence.
-    //
-    let currentStepDescription = 'Failure during ModelInformation registration.';
-    let alreadyTimedOut = false;
-    let registrationTimeout = setTimeout(() => {
-      alreadyTimedOut = true;
-      let errorOnRegistrationSequence = new errors.NotConnectedError(currentStepDescription);
-      return registerCallback(errorOnRegistrationSequence);
-    }, timeoutForDisconnects);
-    this._client.sendEvent(registrationMessage, (registrationError) => {
-      if (alreadyTimedOut) return;
-      if (registrationError) {
-        /* Codes_SRS_NODE_DIGITAL_TWIN_DEVICE_06_011: [Will indicate an error via a callback or by promise rejection if the registration message fails.] */
-        clearTimeout(registrationTimeout);
-        return registerCallback(registrationError);
-      } else {
-        this._setAllInterfaceInstancesRegistered();
-        this._enableAllCommands();
-        currentStepDescription = 'Failure during the retrieval of the device twin.';
-        this._client.getTwin((getTwinError, twinResult) => {
-          if (alreadyTimedOut) return;
-          if (getTwinError) {
-            clearTimeout(registrationTimeout);
-            return registerCallback(getTwinError);
-          } else {
-            this._twin = twinResult as Twin;
-            this._initialWritablePropertyProcessing();
-            //
-            // Intentionally letting a failure of SDK information report
-            // be ignored.
-            //
-            currentStepDescription = 'Failure during the SDK reporting.';
-            this._sdkInformation.language.report('Node.js', (err?: Error) => {
-              if (alreadyTimedOut) return;
-              if (err) {
-                debug('Error updating the SDK language: ' + err.toString());
-              }
-              this._sdkInformation.version.report(packageJson.name + '/' + packageJson.version, (err?: Error) => {
-                if (alreadyTimedOut) return;
-                if (err) {
-                  debug('Error updating the SDK version: ' + err.toString());
-                }
-                this._sdkInformation.vendor.report('Microsoft Corporation', (err?: Error) => {
-                  if (alreadyTimedOut) return;
-                  if (err) {
-                    debug('Error updating the SDK vendor: ' + err.toString());
-                  }
-                  clearTimeout(registrationTimeout);
-                  return registerCallback();
-                });
-              });
-            });
-          }
-        });
-      }
-    });
+    let transport;
+    if (ws) {
+      transport = MqttWs;
+    } else {
+      transport = Mqtt;
+    }
+
+    const client = Client.fromConnectionString(connStr, transport);
+    client.setOptions({ deviceCapabilityModel: '&digital-twin-model-id=' + capabilityModel });
```

Comment:
>'&digital-twin-model-id=' [](start = 47, length = 25)

I probably would have put this down into mqtt-base.

Reply:
ok.

---
In reply to: [401979303](https://github.com/Azure/azure-iot-sdk-node/pull/738#discussion_r401979303) [](ancestors = 401979303)